hal/crc_api.h

```diff
@@ -0,0 +1,117 @@
+#ifndef MBED_CRC_HAL_API_H
```

Comment:
I think that the header file needs to be updated to be consistent with other HAL headers.
Doxygen groups, requirements, undefined behavior, potential bugs sections should be added. Example can be found here:
https://github.com/ARMmbed/mbed-os/blob/feature-hal-spec-ticker/hal/us_ticker_api.h

I have also created some example update with the requirements for testing, which can be found here:
https://github.com/mprse/mbed-os/blob/scartmell-arm_crc_tests/hal/crc_api.h

The requirements were created based on provided function descriptions, so probably they need to be extended. 
Please fill free to use this example in order to update the header file.